modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/DefaultJavaModelLoaderDelegate.java

```diff
@@ -106,6 +106,10 @@ public ExtensionDeclarer declare(ExtensionLoadingContext context) {
             .withCategory(extensionElement.getCategory())
             .withModelProperty(new ExtensionTypeDescriptorModelProperty(extensionElement));
 
+    // TODO MULE-14517: This workaround should be replaced for a better and more complete mechanism
+    context.getParameter("COMPILATION_MODE")
```

Comment:
Use the constant defined in `BaseExtensionResourcesGeneratorAnnotationProcessor.java`

Reply:
can't, BaseExtensionResourcesGeneratorAnnotationProcessor is ext-spring-support, this is ext-support. I could declare the constant here, but then I'd have to make it API and I prefer to keep it all internal since this is a hack and will be removed in 4.2

Reply:
sorry I got confused. dismiss it.